src/main/scala/viper/silver/plugin/standard/inline/InlineRewrite.scala

```diff
@@ -0,0 +1,79 @@
+package viper.silver.plugin.standard.inline
+
+import scala.collection.mutable
+import viper.silver.ast._
+import viper.silver.ast.utility.ViperStrategy
+import viper.silver.ast.utility.rewriter.Traverse
+
+trait InlineRewrite {
+
+  /**
+    * Rewrites a method by expanding the predicates in its pre- and postconditions,
+    * then by removing the corresponding unfold and fold statements in the body.
+    * Note that we only remove unfolds for predicates from the preconditions,
+    * and folds for predicates from the postconditions.
+    * (I think that's how it works.)
+    */
+  def rewriteMethod(method: Method, program: Program): Method = {
+    val (expandedPres, expandablePrePredsIds) = method.pres
+      .map { expandPredicates(_, method, program) }
+      .unzip
+    val (expandedPosts, expandablePostPredsIds) = method.posts
+      .map { expandPredicates(_, method, program) }
+      .unzip
+    val expandablePrePredIds = expandablePrePredsIds.fold(Set())(_ ++ _)
+    val expandablePostPredIds = expandablePostPredsIds.fold(Set())(_ ++ _)
```

Comment:
would calling `.flatten.toSet` do the same thing? Just out of curiosity 🤔

Reply:
Oooh that's very neat! Scala is _so_ nice